### PR TITLE
Riptide

### PR DESCRIPTION
**Pull request by chizzy376**
## About The Pull Request
# full-merge ready edition
Adds a new coastal-themed map to the game, combining a lush jungle with a warm desert. Come take a swim during the winter. (It's pretty much summer for Aussies right now)
THEMAPWOOO

## Why It's Good For The Game
More map good?

## Q&A

Q: Where is X?
A: 
island13

Q: Why do certain parts of the map feel blank?

A: I don't have the assets and would like to mobilize YOU, the player, to make good coastal-themed sprites for mappers to use. Tiki torches, beach chairs, umbrellas, coconuts, palm trees, you name it! 

Q: Are you sure X is a good spot?

A: Nope! This is all subject to change.

Q: Weedable grass?

A: Yes.

-----------
Credits to:
/tg/ for fishing sprites
doomvoyager for sandman + coconut drink sprites
Mack (Aka Married2DaGame) for being the co-producer
Grayson
5 cups of coffee at 4 AM
## Changelog
:cl:
add: New coastal map - Riptide.
/:cl:

### CODE REVIEW

**Comment by chizzy376** [comment]
oh you fuck

**Comment by chizzy376** [comment]
YEAH, that's what I thought. I've been through this rodeo twice already, you can't stop me bot.

**Comment by FosterGrime-Heymynameslouie** [comment]
beach episode

**Comment by chizzy376** [comment]
> beach episode

Some could say this is finally the anime's fan service

**Comment by zzurge** [comment]
time for some fun in the sun

**Review by RipGrayson — CHANGES_REQUESTED** [review]
It seems that your LZs don't have any shutters at all, this is fairly standard to keep xenos from camping the ship before it arrives. LZ1 also seems to lack wiring and an APC, I'm not sure if this is intentional.

**Review by RipGrayson — CHANGES_REQUESTED** [review]
rivbar
rivbarge
These two disk candidates seem to lack APCs or wiring. This makes it impossible for xenos to turn the disk off once it's been started.

**Comment by chizzy376** [comment]
> It seems that your LZs don't have any shutters at all, this is fairly standard to keep xenos from camping the ship before it arrives. LZ1 also seems to lack wiring and an APC, I'm not sure if this is intentional.

OH I KNEW I FORGOT SOMETHING

**Comment by ilikesaying** [comment]
Beach Episode!

**Comment by Oscari-pog** [comment]
Wooden ship seems comically defensible

Yes even if you melt all the sandbags and the mounted machinegun

**Comment by TiviPlus** [comment]
The blockers/killzone whichever that is being spammed around the islands (which also have wide open silos on them) seems very annoying

**Comment by chizzy376** [comment]
> The blockers/killzone whichever that is being spammed around the islands (which also have wide open silos on them) seems very annoying

Hello, blockers only block movement but not projectiles. This is to prevent players from stepping into the instant death sea tiles.

**Comment by TiviPlus** [comment]
How about removing the instant kill tiles instead of finding ways to make corridors even more marine sided (and then making silos spawn there)

**Comment by chizzy376** [comment]
> How about removing the instant kill tiles 

No, you can't access them thanks to the blockers.

> Ways to make corridors even more marine sided

They're bridges, sir, you don't have to go on them since there's no objectives there. 

> Making silos spawn there

Implying crash was ever taken into account by anyone

**Comment by Aporhtonoma** [comment]
Id expect nothing less from the creator of slumbridge gods greatest map on tgmc please speedmerge

File: _maps/riptide.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+    "map_name": "riptide",
```
Comment (vvvv-vvvv) [inline _maps/riptide.json:2]:
```suggestion
    "map_name": "Riptide",
```

File: _maps/riptide.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    "map_path": "map_files/riptide",
+    "map_file": "riptide.dmm",
+	"disk_sets": {
+		"set1": 1,
+		"set2": 1,
+		"set3": 1
+	},
```
Comment (vvvv-vvvv) [inline _maps/riptide.json:9]:
```suggestion
    "disk_sets": {
        "set1": 1,
        "set2": 1,
        "set3": 1
    },
```

File: _maps/riptide.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    "armor": "black",
+	"quickbuilds": 2000,
+	"announce_text": "We have received a faint distress signal originating from a sandy island with a dense jungle. The ship has jumped within range of what looks like a beach resort. TGMC, gear up, don't forget your sunscreen and get ready to respond!",
+	"traits":[{
+	}]
```
Comment (vvvv-vvvv) [inline _maps/riptide.json:14]:
```suggestion
    "quickbuilds": 2000,
    "announce_text": "We have received a faint distress signal originating from a sandy island with a dense jungle. The ship has jumped within range of what looks like a beach resort. TGMC, gear up, don't forget your sunscreen and get ready to respond!",
    "traits": [{}]
```

File: code/game/objects/effects/misc.dm [hunk 4]
```diff
@@ -71,6 +71,14 @@
 	if(icon_state == "blocker")
 		icon_state = ""
 
+/obj/effect/forcefield/dense0forguns
+	resistance_flags = PROJECTILE_IMMUNE
```
Comment (vvvv-vvvv) [inline code/game/objects/effects/misc.dm:75]: what is this for? the parent can block projectiles if that's what you want

File: code/game/objects/effects/misc.dm [hunk 5]
```diff
@@ -71,6 +71,14 @@
...
+
+/obj/effect/forcefield/dense0forguns/Initialize(mapload)
+	. = ..()
+	if(icon_state == "blocker")
+		icon_state = ""
```
Comment (vvvv-vvvv) [inline code/game/objects/effects/misc.dm:80]:
```suggestion
```

**Review by vvvv-vvvv — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/misc.dm (hunk 4 above)
Comment (chizzy376) [reply code/game/objects/effects/misc.dm:75]: It makes projectiles pass through it

File: code/game/objects/effects/misc.dm (hunk 5 above)
Comment (chizzy376) [reply code/game/objects/effects/misc.dm:80]: Doesn't removing this make the special forcefield visible ingame?

Comment (vvvv-vvvv) [reply code/game/objects/effects/misc.dm:80]: no, it already gets inherited from parent

**Comment by chizzy376** [comment]
island9caves
New caves update, can't post in PR body as image would exceed 10MB

**Comment by chizzy376** [comment]
Oh god what did i press

**Comment by Arturlang** [comment]
Please poplock this in crash or way more silos in better places

**Comment by chizzy376** [comment]
> Please poplock this in crash 

I'd rather not do that

> or way more silos in better places

Tell me where, I think the silos that I placed are in good locations and allow xenos to stage offensives without being 5 tiles away from crash site nor disk (Unlike icy caves)

**Comment by Arturlang** [comment]
> > Please poplock this in crash
> 
> I'd rather not do that
> 
> > or way more silos in better places
> 
> Tell me where, I think the silos that I placed are in good locations and allow xenos to stage offensives without being 5 tiles away from crash site nor disk (Unlike icy caves)
The center is completely barren of any silos, and the bottom left silo is pretty much useless as the pheremone range just barely extends to the closest building north of it

**Comment by chizzy376** [comment]
> The center is completely barren of any silos, and the bottom left silo is pretty much useless as the pheremone range just barely extends to the closest building north of it. A lot of disks just require you to rely on a shrike or make a long trek to a silo just to heal up. Maps like ice caves or desparity? i think also give you places with double phero tower coverage in some places, whereas for this map most of it has no pheros, and the silos are intentionally placed to require extra traveling to reach. Not to mention this map has _very_ few silos for how big it is compared to other maps

Thank you for the feedback! Could you send an image as to where exactly you'd like to see new silos placed?

**Comment by Loafin34** [comment]
from what i've seen, marine winrate on this map is 70-80% because of how easy it is to travel using the river, which is maze/weed free. Xenos have to pick a long-range caste like carrier or warlock to win. I main marine and I see the xenos being unable to attack.

**Comment by Yttriums** [comment]
atmos pipes on this map should really be connected between buildings as they are on other maps. really shuts down xeno mobility to have areas like medbay completely isolated

**Comment by lKiseki** [comment]
image

but why tho

**Comment by lKiseki** [comment]
image

i also have no issues dropping this here if you wanna take feedback like this

**Comment by chizzy376** [comment]
how do i block someone on github

**Comment by chizzy376** [comment]
> image


cope

**Comment by MiniMeatwad** [comment]
sharp words

**Comment by Ragecrabbo** [comment]
Very visually interesting map, i especially like the cave structures and nuke disk boat. 
One small issue i noticed while playing though is that the northern area tends to be fairly easy for marines to hold after taking it from xenos, mainly due to close proximity to water but also because of its openness. 
While the former is more of a feature than a problem i would suggest adding a few walls or a shack or some other form of cover somewhere in the red squared area to break off marine sightlines a bit
map
Overall though this is a cool unique map that is fun to play especially on crash and i hope you continue to work and improve on it.

**Comment by chizzy376** [comment]
> Very visually interesting map, i especially like the cave structures and nuke disk boat. One small issue i noticed while playing though is that the northern area tends to be fairly easy for marines to hold after taking it from xenos, mainly due to close proximity to water but also because of its openness. While the former is more of a feature than a problem i would suggest adding a few walls or a shack or some other form of cover somewhere in the red squared area to break off marine sightlines a bit map Overall though this is a cool unique map that is fun to play especially on crash and i hope you continue to work and improve on it.

More shacks sounds perfect, thank you for the feedback!

**Comment by RipGrayson** [comment]
Posting feedback from Discord
deadeagle1
deadeagle2

**Comment by chizzy376** [comment]
Nuh uh, not stale, I'm working on a big update

**Comment by RipGrayson** [comment]
I don't know the status of this, but I'm going to reopen it once more. If it hasn't received an update by the time of next staling out I'll let it die.

**Comment by chizzy376** [comment]
I'm very sorry for disappointing you all. I will finish it someday.

**Pull request closed without merging**